client/asset/btc/btc.go

```diff
@@ -2638,7 +2638,13 @@ func (btc *ExchangeWallet) SwapConfirmations(_ context.Context, id dex.Bytes, co
 	if err != nil {
 		return 0, false, err
 	}
-	return btc.node.swapConfirmations(txHash, vout, pkScript, startTime)
+	confs, spent, err := btc.node.swapConfirmations(txHash, vout, pkScript, startTime)
+	if _, isRPC := btc.node.(*rpcClient); isRPC && errors.Is(err, asset.CoinNotFoundError) {
+		// We'll assume caller is silent for this error, but it is a bad
+		// indication for full node wallets, so let's make extra noise here.
+		btc.log.Warnf("Failed to find swap transaction %v.", txHash)
+	}
+	return confs, spent, err
```

Comment:
This is what bugs me about this approach of logging in the asset backend.  Logging an error or a warning is strange because we are about to return an error, so the question is why doesn't the error just convey all the needed info.

This leads me to other ideas for work on the next release cycle:
- `asset.UnexpectedCoinNotFoundError` (or `asset.ExpectedCoinNotFoundError`) vs `asset.CoinNotFoundError`
-  return a nil error for not found scenarios, and instead have a -1 confs or a `found bool` return